### DISCUSSION TITLE
Deploying the app on cloudflare [Question]

### DISCUSSION THREAD

**Discussion by Johnynight** [flet-dev/flet#1726] [Q&A]
### Question

Why is it that when I deploy an application on cloudflare I have an endless download? I did everything as in the instructions https://flet.dev/docs/guides/python/publishing-static-website#cloudflare-pages and the project in the example is also in the download https://todo-2.pages.dev

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Thanks. There really seems to be an issue.
Let me look at it.

**Comment by ndonkoHenri** [comment]
@FeodorFitsner, have a look:
![image](https://github.com/flet-dev/flet/assets/98978078/460a4156-630a-4d05-ae43-9f2370079686)

**Reply by ndonkoHenri** [reply]
By the way Feodor, you forgot to document these two props!

**Reply by FeodorFitsner** [reply]
Yeah, the problem is that app was published with Flet pre-0.9.0 and in runtime it loads the latest Flet 0.9.0 as version is not pinned.
Add [`requirements.txt`](https://flet.dev/docs/guides/python/publishing-static-website#loading-packages) to your app with Flet version pinned:

```
flet==0.9.0
```

and republish the app.

**Reply by Johnynight** [reply]
Is this an intractable problem?

**Reply by FeodorFitsner** [reply]
I don't think so. You are welcome to create a new issue to improve that.

**Reply by ndonkoHenri** [reply]
@FeodorFitsner your suggestion didn't work. 
My app(flutils) just had `flet` in the requirements and the `v0.9.0` of flet was loaded at runtime and not the pre (confirmation was done by looking at the `pre` variable in the `index.html` \ I also didn't set the `--pre` flag in my publish command).

So, setting `flet==0.9.0` in my requirements, as suggested by Feodor showed thesame result(see image in my previous comment) as without.

The solution to this problem was to simply set `flet<=0.9.0` in my `requirements.txt` and my app was shown successfully.
See https://flutils.pages.dev

**What is the issue with v0.9.0?**
The introduction of `page.debug` and `platform_brightness` is the actual cause of this issue. Feodor, please have a closer look at the image in my previous comment, and you will certainly be able to catch the bug.

**Reply by FeodorFitsner** [reply]
Will take a look, thanks for reporting that.

**Reply by FeodorFitsner** [reply]
IDK, static publishing works for me. I have just `flet-pyodide` in `requirements.txt`. What errors do you have in browser's console?

![image](https://github.com/flet-dev/flet/assets/5041459/7ad9009f-67a3-4168-a5e8-f35198ee18a0)

**Reply by ndonkoHenri** [reply]
Maybe the error is shown only on deploy? 😕 

Anyway, I got the errors below while looking at the console of my deployed app back then. Towards the end, you see `debug` and `platform_brightness` being mentioned.

![image](https://github.com/flet-dev/flet/assets/98978078/460a4156-630a-4d05-ae43-9f2370079686)

**Reply by ndonkoHenri** [reply]
Open the console on [this](https://flet-material3-demo.pages.dev/) deployed app to see the errors.

**Reply by FeodorFitsner** [reply]
Re-publish the app with 0.9.0. It's not published with 0.9.0. `pre` is disabled - good, it pulls flet-pyodide 0.9.0 while running (checked on Network tab) - good, but it doesn't look like the app is published with 0.9.0.
